### PR TITLE
[PM-4531] Do not override webauthn on excluded domains

### PR DESCRIPTION
**Pull request by coroiu**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective
- Add support for excluded domains
- Fix crash when using `localhost` (https://bitwarden.atlassian.net/browse/PM-4573)

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team
- Ensure that all UI additions follow [WCAG AA requirements](https://contributing.bitwarden.com/contributing/accessibility/)

### CODE REVIEW

File: libs/common/src/vault/services/fido2/domain-utils.ts [hunk 1]
```diff
@@ -5,7 +5,9 @@ export function isValidRpId(rpId: string, origin: string) {
   const parsedRpId = parse(rpId, { allowPrivateDomains: true });
 
   return (
-    parsedOrigin.domain === parsedRpId.domain &&
-    parsedOrigin.subdomain.endsWith(parsedRpId.subdomain)
+    (parsedOrigin.domain === parsedRpId.domain &&
```
Comment (shane-melton) [inline libs/common/src/vault/services/fido2/domain-utils.ts:8]: question: I'm still new to this logic, but this change looks to ignore the `parsedOrigin.domain === parsedRpId.domain` check if `paredOrigin.subdomain == null`. Is skipping the `domain` check intended when `subdomain` is null?

**Review by shane-melton — COMMENTED** [review]
Just had one question before giving approval, nice work!

File: libs/common/src/vault/services/fido2/domain-utils.ts (hunk 1 above)
Comment (coroiu) [reply libs/common/src/vault/services/fido2/domain-utils.ts]: Argh, parantheses! Fixed!

**Review by shane-melton — APPROVED** [review]
Thanks for the fix and additional tests!

**Review by LRNcardozoWDF — APPROVED** [review]

**Pull request merged**